include/openPMD/backend/BaseRecordComponent.hpp

```diff
@@ -59,8 +59,25 @@ class BaseRecordComponent : public Attributable
      */
     bool constant() const;
 
-OPENPMD_protected:
-    BaseRecordComponent();
+    /**
+     * Get data chunks that are available to be loaded from the backend.
+     * Note that this is backend-dependent information and the returned
+     * information may hence differ between different backends:
+     * * The ADIOS backends (versions 1 and 2) will return those chunks that
+     *   the writer has originally written.
+     * * The JSON backend will reconstruct the chunks by iterating the dataset.
+     * * The HDF5 backend will return the whole dataset as one large chunk.
+     *
+     * The results depend solely on the backend and are independent of any
+     * openPMD-related information. Especially is this call explicitly unrelated
+     * to openPMD's concept of particle patches, which users may additionally
+     * wish to use to store user-defined, backend-independent chunking
+     * information on particle datasets.
```

Comment:
The description is fitting for the way that things are implemented right now. So, if changing this, should we also change the implementation? To what?

We could maybe think about doing a more high-level approach in a fallthrough manner:

1. Query available chunks from backend
2. If the backend doesn't know, query particle patches (ordering? this first?)
3. If none of this works, return whole dataset as one chunk.

Reply:
I meant that the formulation `Especially is this call explicitly unrelated to openPMD's concept of particle patches` can be changed to `Note that this call currently does not take into account the openPMD concept of particle patches...`.

Reply:
> currently does not take into account the openPMD concept of particle patches

Alright, that's an easier fix for sure. 

But the question still stands whether it would be wise to mix both notions one day. What if both particle patches and backend-specific chunks are available on a dataset? Then..
1. … both information may contradict each other. Especially given the clear semantical gap between chunks in HDF5 and particle patches, they're not even trying to achieve the same thing.
2. … we would need to pick one of both to return to the user. It's not really our place to make that decision because depending on the use case, either of both may be more interesting:
     * If optimizing for efficiency, throughput and data distribution, loading data in a way suggested by the backend (and with knowledge on writing ranks) is to be preferred. 
     * If interested in physical quantities, such as the hyperrectangle containing a particle path, loading the particle patches is to be preferred.

This is not saying that I am totally against it, but I think that those points should be considered. Rough draft:
1. If a user explicitly defines particle patches, this should count as explicitly overwriting the chunk information given implicitly by the backend. So *if* we decide to prioritize one over the other, we should probably prioritize the particle patches over the backend-dependent info.
2. If we prioritize particle patches, they should in general carry information at least as useful as the backend-dependent chunk info. So, standardize some kind of `writerID` attribute.
3. Since both notions of chunks may and will differ from one another, allow users to explicitly prefer the backend-dependent information if they wish. Something like `::availableChunks( ChunksInfoSource = ChunksInfoSource::ParticlePatches )` could do.

Reply:
I concur with your analysis. Let's cross that bridge when we come to this, in practice we need particle patches for checkpoint-restart workflows and selection workflows to avoid iterating the whole particle table. It also served as a good work-around for strided HDF5 data, but that's not a primary concern if we have backend info, i.e. in ADIOS2.

Can we discuss this and questions on `writerID` in a VC? I have some experiences on how useful/not useful that would be in practice and need to fully understand how you want to use it.

Reply:
Linking PR #824 here for an example on how I am intending to use this. Tldr: `writerID` by itself is only an identifier that we need to make sense of by adding meta information, such as e.g. hostnames, or similar information that we may wish to rely upon for chunk distribution strategies.